releasey/06-build-and-stage-docker-images.sh

```diff
@@ -0,0 +1,137 @@
+    - DockerHub credentials must be configured (docker login)
+    - Must be run from a release tag (apache-polaris-x.y.z-incubating-rcN)
+
+  Options:
+    -h --help
+        Print usage information.
+
+  Examples:
+    $(basename "$0")
+
+EOF
+}
+
+ensure_cwd_is_project_root
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --help|-h)
+      usage
+      exit 0
+      ;;
+    *)
+      print_error "Unknown option/argument $1"
+      usage >&2
+      exit 1
+      ;;
+  esac
+done
+
+# Determine version from current git tag
+print_info "Determining version from current git tag..."
+
+if ! git_tag=$(git describe --tags --exact-match HEAD 2>/dev/null); then
+  print_error "Current HEAD is not on a release tag. Please checkout a release tag first."
+  print_error "Use: git checkout apache-polaris-x.y.z-incubating-rcN"
+  exit 1
+fi
+print_info "Found git tag: ${git_tag}"
+
+# Extract version components from git tag in one regex match
+git_tag_regex="^apache-polaris-([0-9]+)\.([0-9]+)\.([0-9]+)-incubating-rc([0-9]+)$"
+if [[ ! ${git_tag} =~ ${git_tag_regex} ]]; then
+  print_error "Invalid git tag format: ${git_tag}"
+  print_error "Expected format: apache-polaris-x.y.z-incubating-rcN"
+  exit 1
+fi
+
+# Extract version components from regex match
+major="${BASH_REMATCH[1]}"
+minor="${BASH_REMATCH[2]}"
+patch="${BASH_REMATCH[3]}"
+rc_number="${BASH_REMATCH[4]}"
+version="${major}.${minor}.${patch}-incubating"
+docker_tag="${version}-rc${rc_number}"
+
+print_info "Starting Docker image build and staging process..."
+print_info "Version: ${version}"
+print_info "RC number: ${rc_number}"
+print_info "Docker tag: ${docker_tag}"
+echo
+
+# Build and push polaris-server Docker image
+print_info "Building and pushing polaris-server Docker image..."
+exec_process ./gradlew :polaris-server:assemble :polaris-server:quarkusAppPartsBuild --rerun \
+  -Dquarkus.container-image.build=true \
+  -Dquarkus.container-image.push=true \
+  -Dquarkus.docker.buildx.platform="linux/amd64,linux/arm64" \
+  -Dquarkus.container-image.tag="${docker_tag}"
+
+# Build and push polaris-admin Docker image
+print_info "Building and pushing polaris-admin Docker image..."
+exec_process ./gradlew :polaris-admin:assemble :polaris-admin:quarkusAppPartsBuild --rerun \
+  -Dquarkus.container-image.build=true \
+  -Dquarkus.container-image.push=true \
+  -Dquarkus.docker.buildx.platform="linux/amd64,linux/arm64" \
+  -Dquarkus.container-image.tag="${docker_tag}"
+
+echo
+print_success "🎉 Docker images built and staged successfully!"
+echo
+print_info "Published Docker images:"
+print_info "- polaris-server: ${DOCKER_REGISTRY}/apache/polaris:${docker_tag}"
+print_info "- polaris-admin: ${DOCKER_REGISTRY}/apache/polaris-admin-tool:${docker_tag}"
+echo
+print_info "Docker Hub links:"
+print_info "- ${DOCKER_HUB_URL}/r/apache/polaris/tags"
+print_info "- ${DOCKER_HUB_URL}/r/apache/polaris-admin-tool/tags"
+echo
+print_info "Next steps:"
+print_info "1. Start the vote thread"
```

Comment:
I think it's worth to have a proposal for the subject and body of the email printed here.
Actually, maybe even in it's own shell script.

Reply:
Actually, I'm thinking of removing the `releaseEmailTemplate` (and related) tasks and do that in the release scripts. Mean, the scripts already have all the information, so the indirection via Gradle, which has to figure out the details again, isn't necessary.